buildpackrunner/runner.go

```diff
@@ -95,55 +115,45 @@ func (runner *Runner) Run() error {
 
 	tarPath, err := exec.LookPath("tar")
 	if err != nil {
-		return err
-	}
-
-	//prepare the final droplet directory
-	contentsDir, err := ioutil.TempDir("", "contents")
-	if err != nil {
-		return newDescriptiveError(err, "Failed to create droplet contents dir")
+		return "", err
 	}
 
 	//generate staging_info.yml and result json file
-	infoFilePath := path.Join(contentsDir, "staging_info.yml")
+	infoFilePath := path.Join(runner.contentsDir, "staging_info.yml")
 	err = runner.saveInfo(infoFilePath, detectedBuildpack, detectOutput, releaseInfo)
 	if err != nil {
-		return newDescriptiveError(err, "Failed to encode generated metadata")
-	}
-
-	appDir := path.Join(contentsDir, "app")
-	err = runner.copyApp(runner.config.BuildDir(), appDir)
-	if err != nil {
-		return newDescriptiveError(err, "Failed to copy compiled droplet")
+		return "", newDescriptiveError(err, "Failed to encode generated metadata")
 	}
 
-	err = os.MkdirAll(path.Join(contentsDir, "tmp"), 0755)
-	if err != nil {
-		return newDescriptiveError(err, "Failed to set up droplet filesystem")
+	for _, name := range []string{"tmp", "logs"} {
```

Comment:
Those directories are created for the droplet after all the buildpacks have run. We feel that creating the directories in this location matches the separation of concerns. If you feel strongly otherwise, we're happy to change